Actions/ReadSettings/ReadSettings.ps1

```diff
@@ -141,13 +141,20 @@ try {
                 if ($ghEnvironment) {
                     $branchPolicy = ($ghEnvironment.protection_rules | Where-Object { $_.type -eq "branch_policy" })
                     if ($branchPolicy) {
-                        Write-Host "GitHub Environment $envName has branch policies, getting branches from GitHub API"
-                        $branchesUrl = "$($ENV:GITHUB_API_URL)/repos/$($ENV:GITHUB_REPOSITORY)/environments/$([Uri]::EscapeDataString($envName))/deployment-branch-policies"
-                        Write-Host "Getting branches for $envName from GitHub API"
-                        $branches = @((InvokeWebRequest -Headers $headers -Uri $branchesUrl -ignoreErrors | ConvertFrom-Json).branch_policies | ForEach-Object { $_.name })
+                        $branches = @()
```

Comment:
Nothing after line 102 should run as part of the "Pull Request Build" (the original issue) as there is no _Deploy_ job there. Please address.

Reply:
And if I haven't said it enough lately: considering extracting an action (`DetermineEnvironments`) action 😄 
This part goes beyond reading settings.

Reply:
I really don't believe in refactoring during bug fixing (which I think I have said a number of times as well).
I am totally aware that getting environments should be factored out of ReadSettings, but I think that refactoring should be done as a refactoring task in order to do it right.

Reply:
but... - I will move it out into a function with more clearance on what and how.